src/main/java/com/aws/greengrass/certificatemanager/CertificateManager.java

```diff
@@ -114,30 +120,55 @@ public String getCaPassPhrase() {
      * @param cb  Certificate consumer
      * @throws KeyStoreException if unable to access KeyStore
      * @throws CsrProcessingException if unable to process CSR
+     * @throws RuntimeException for any unknown runtime error
      */
-    @SuppressWarnings("PMD.AvoidCatchingGenericException")
+    @SuppressWarnings({"PMD.AvoidCatchingGenericException", "PMD.ExceptionAsFlowControl"})
     public void subscribeToServerCertificateUpdates(@NonNull String csr, @NonNull Consumer<X509Certificate> cb)
             throws KeyStoreException, CsrProcessingException {
-        // BouncyCastle can throw RuntimeExceptions, and unfortunately it is not easy to detect
-        // bad input beforehand. For now, just catch and re-throw a CsrProcessingException
+        // deduplicate subscriptions with the same callback and csr
```

Comment:
Thoughts on ignoring the CSR and keying off just the callback? Technically the CSR gets re-generated on each request. It should be deterministic, but it still feels less right than keying off just the callback.

This API is poorly designed all around, so I think we'll want to move away from it anyway. I think I'd rather enforce 1 callback per cert and avoid potential issues with the CSR changing, especially since I can't think of a good way for a caller to actually use this API to generate multiple server certs with the same callback.

Reply:
Yes, that seems perfectly reasonable to me